xcode/Subconscious/Shared/Services/DatabaseService.swift

```diff
@@ -257,6 +257,52 @@ final class DatabaseService {
         return results.first?.col(0)?.toInt()
     }
     
+    func listEntries(for slashlinks: [Slashlink]) throws -> [EntryStub] {
+        return try slashlinks.compactMap({ slashlink in
+            return try readEntry(for: slashlink)
+        })
+    }
+    
+    func readEntry(for slashlink: Slashlink) throws -> EntryStub? {
+        guard self.state == .ready else {
+            throw DatabaseServiceError.notReady
+        }
+
+        let results = try database.execute(
+            sql: """
+        SELECT id, modified, excerpt
```

Comment:
We will need to update this query for the new database structure introduced in #561.